app/src/test/java/com/github/onlynotesswent/model/flashcard/FlashcardRepositoryFirestoreTest.kt

```diff
@@ -0,0 +1,370 @@
+    `when`(mockResolveTask.addOnSuccessListener(any())).thenAnswer { invocation ->
+      val listener = invocation.arguments[0] as com.google.android.gms.tasks.OnSuccessListener<Void>
+      listener.onSuccess(null)
+      mockResolveTask
+    }
+    `when`(mockResolveTask.addOnFailureListener(any())).thenReturn(mockResolveTask)
+
+    // Mock the behavior of the DocumentReference get operation
+    `when`(mockDocumentReference.get()).thenReturn(mockDocumentTask)
+    `when`(mockDocumentTask.addOnSuccessListener(any())).thenAnswer { invocation ->
+      val listener =
+          invocation.arguments[0]
+              as com.google.android.gms.tasks.OnSuccessListener<DocumentSnapshot>
+      listener.onSuccess(mockDocumentSnapshot)
+      mockDocumentTask
+    }
+    `when`(mockDocumentTask.addOnFailureListener(any())).thenReturn(mockDocumentTask)
+
+    // Mock the DocumentSnapshot to return expected fields for a Flashcard
+    `when`(mockDocumentSnapshot.exists()).thenReturn(true)
+    `when`(mockDocumentSnapshot.id).thenReturn(flashcard.id)
+    `when`(mockDocumentSnapshot.getString("front")).thenReturn(flashcard.front)
+    `when`(mockDocumentSnapshot.getString("back")).thenReturn(flashcard.back)
+    `when`(mockDocumentSnapshot.getTimestamp("nextReview")).thenReturn(flashcard.nextReview)
+    `when`(mockDocumentSnapshot.getString("userId")).thenReturn(flashcard.userId)
+    `when`(mockDocumentSnapshot.getString("folderId")).thenReturn(flashcard.folderId)
+    `when`(mockDocumentSnapshot.getString("noteId")).thenReturn(flashcard.noteId)
+  }
+
+  @Test
+  fun init_callsAuthStateListener() {
+    val mockAuth = mock(FirebaseAuth::class.java)
+    val mockUser = mock(FirebaseUser::class.java)
+    `when`(mockAuth.currentUser).thenReturn(mockUser)
+
+    var onSuccessCalled = false
+    flashcardRepositoryFirestore.init(mockAuth) { onSuccessCalled = true }
+    assert(onSuccessCalled)
+  }
+
+  @Test
+  fun documentSnapshotToFlashcardConvertsToFlashcard() {
+    `when`(mockDocumentSnapshot.id).thenReturn(flashcard.id)
+    `when`(mockDocumentSnapshot.getString("front")).thenReturn(flashcard.front)
+    `when`(mockDocumentSnapshot.getString("back")).thenReturn(flashcard.back)
+    `when`(mockDocumentSnapshot.getTimestamp("nextReview")).thenReturn(flashcard.nextReview)
+    `when`(mockDocumentSnapshot.getString("userId")).thenReturn(flashcard.userId)
+    `when`(mockDocumentSnapshot.getString("folderId")).thenReturn(flashcard.folderId)
+
+    val convertedFlashcard =
+        flashcardRepositoryFirestore.documentSnapshotToFlashcard(mockDocumentSnapshot)
+
+    assert(convertedFlashcard?.id == flashcard.id)
+    assert(convertedFlashcard?.front == flashcard.front)
+    assert(convertedFlashcard?.back == flashcard.back)
+    assert(convertedFlashcard?.nextReview == flashcard.nextReview)
+    assert(convertedFlashcard?.userId == flashcard.userId)
+    assert(convertedFlashcard?.folderId == flashcard.folderId)
```

Comment:
You could replace this asserts by `assertEquals(convertedFlashcard, flashcard)` which tests equality for all fields and shows neatly the differences in the error logs. 

To see if the "==" is correct, add unit tests for the flashcard data class in a new file FlashcardTest.kt. The responsibility of making sure the equals operator works is not the job of FlashcardRepositoryFirestoreTest.kt but FlashcardTest.kt